src/Service/VendorService/OverDrive/OverDriveBooksVendorService.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * @file
+ * Service for updating book covers from OverDrive.
+ */
+
+namespace App\Service\VendorService\OverDrive;
+
+use App\Exception\IllegalVendorServiceException;
+use App\Exception\UnknownVendorServiceException;
+use App\Service\VendorService\AbstractBaseVendorService;
+use App\Service\VendorService\OverDrive\Api\Client;
+use App\Service\VendorService\ProgressBarTrait;
+use App\Utils\Message\VendorImportResultMessage;
+use App\Utils\Types\IdentifierType;
+use Doctrine\ORM\EntityManagerInterface;
+use GuzzleHttp\ClientInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Class OverDriveBooksVendorService.
+ */
+class OverDriveBooksVendorService extends AbstractBaseVendorService
+{
+    use ProgressBarTrait;
+
+    protected const VENDOR_ID = 14;
+
+    private $apiClient;
+    private $httpClient;
+
+    /**
+     * OverDriveBooksVendorService constructor.
+     *
+     * @param eventDispatcherInterface $eventDispatcher
+     *   Dispatcher to trigger async jobs on import
+     * @param entityManagerInterface $entityManager
+     *   Doctrine entity manager
+     * @param loggerInterface $statsLogger
+     *   Logger object to send stats to ES
+     * @param ClientInterface $httpClient
+     *   Http client to send api requests
+     * @param Client $apiClient
+     *   Api client for the OverDrive API
+     */
+    public function __construct(EventDispatcherInterface $eventDispatcher, EntityManagerInterface $entityManager, LoggerInterface $statsLogger, ClientInterface $httpClient, Client $apiClient)
+    {
+        parent::__construct($eventDispatcher, $entityManager, $statsLogger);
+
+        $this->httpClient = $httpClient;
+        $this->apiClient = $apiClient;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load(): VendorImportResultMessage
+    {
+        if (!$this->acquireLock()) {
+            return VendorImportResultMessage::error(parent::ERROR_RUNNING);
+        }
+
+        $this->loadConfig();
+
+        try {
+            $this->progressStart('Starting eReolen Global import from overdrive API');
+
+            $totalCount = (0 !== $this->limit) ? $this->limit : $this->apiClient->getTotalProducts();
```

Comment:
Should there be a check to ensure that `getTotalProducts` returns an number to ensure that the `do` loop below will terminate.

Reply:
Fixed